packages/core/src/scp-cf/destination-accessor.ts

```diff
@@ -482,19 +285,346 @@ const emptyDestinationByType: DestinationsByType = {
+        );
+      }
+      return getDestinationWithAuthTokens(
+        this.name,
+        this.options.userJwt,
+        this.options
+      );
+    }
+  }
+
+  private async tryForCert(): Promise<Destination | undefined> {
+    const destination = await this.tryService();
+    if (
+      destination &&
+      destination.authentication === 'ClientCertificateAuthentication'
+    ) {
+      return getDestinationWithCertificates(
+        this.name,
+        this.decodedUserJwt,
+        this.options
+      );
+    }
+  }
+
+  private async addProxyConfiguration(
+    destination: Destination | undefined
+  ): Promise<Destination | undefined> {
+    if (!destination) {
+      return;
+    }
+
+    switch (proxyStrategy(destination)) {
+      case ProxyStrategy.ON_PREMISE_PROXY:
+        return addProxyConfigurationOnPrem(destination, this.options.userJwt);
+      case ProxyStrategy.INTERNET_PROXY:
+        return addProxyConfigurationInternet(destination);
+      case ProxyStrategy.NO_PROXY:
+        return destination;
+      default:
+        throw new Error(
+          'Illegal argument: No valid proxy configuration found in the destination input to be aded.'
+        );
+    }
+  }
+
+  private async processDestinationResult(
+    destination: Destination | undefined
+  ): Promise<Destination | undefined> {
+    if (!destination) {
+      return;
+    }
+
+    logger.info('Successfully retrieved destination from destination service.');
+
+    return this.updateCacheSingle(
+      await this.addProxyConfiguration(destination)
+    );
+  }
+
+  private updateCacheService(
+    token: DecodedJWT,
+    destinations: DestinationsByType
+  ) {
+    if (this.useCache) {
+      destinationCache.cacheRetrievedDestinations(
+        token,
+        destinations,
+        this.isolationStrategy
+      );
+    }
+  }
+
+  private updateCacheSingle(
+    destination: Destination | undefined
+  ): Destination | undefined {
+    if (!this.useCache || !destination) {
+      return destination;
+    }
+    // The information if the destination originates from a instance of account service is not relevant for the caching.
+    destinationCache.cacheRetrievedDestinations(
+      this.decodedUserJwt || this.decodedProviderToken,
+      { instance: [], subaccount: [destination] },
+      this.isolationStrategy
+    );
+    return destination;
+  }
+
+  private async tryService(): Promise<Destination | undefined> {
+    const allDest: AllDestinations = {
+      provider: await this.getProviderDestinations(),
+      subscriber: await this.getSubscriberDestination()
+    };
+    const selectedFromService = await this.selectionStrategy(
+      allDest,
+      this.name
+    );
+
+    if (!selectedFromService) {
+      logger.info('Could not retrieve destination from destination service.');
+    }
+    return selectedFromService ?? undefined;
+  }
+
+  private async destinationFoundFromSubscriber(): Promise<boolean> {
```

Comment:
This function is only used to check whether there are subscriber destinations for the subscriber first strategy, right? Is it even necessary to execute the selection strategy here? (If yes, I guess a comment would be helpful).
Consider something like this: `canFindSubscriberDestination`

Reply:
yes you need to execute the selection strategy so that you know if you would find a dest on the subscriber result.